code/mission/import/xwingmissionparse.cpp

```diff
@@ -623,6 +623,208 @@ void parse_xwi_flightgroup(mission *pm, const XWingMission *xwim, const XWMFligh
+	// NOTE: Y and Z are swapped after all operartions are perfomed
+    // units are in kilometers (after processing by xwinglib which handles the factor of 160), so scale them up
+	float objectPosX = oj->object_x*1000;
+	float objectPosY = oj->object_y*1000;
+	float objectPosZ = oj->object_z*1000;
+
+	int mine_dist = 400; // change this to change the distance between the mines
+	/** The minefield is a square 2d along two planes (a,b) centred on the given position **/
+	float offsetAxisA = 0 -(mine_dist / 2 * (number_of_objects - 1)); // (- the distance to centre the grid)
+	float offsetAxisB = 0 -(mine_dist / 2 * (number_of_objects - 1));
+		
+	matrix orient;
+	xwi_determine_object_orient(&orient, oj);
+
+	// Copy objects in Parse_objects to set for name checking below
+	// This only needs to be done fully once per object group then can be added to after each new object
+	SCP_set<SCP_string> objectNameSet;
+	for (int n = 0; n < Parse_objects.size(); n++) {
+		objectNameSet.insert(Parse_objects[n].name);
+	}
+
+	// now begin to configure each object in the group (mines multiple)
+
+	for (int a = 0; a < number_of_objects; a++) { // make an a-b 2d grid from the mines
+		offsetAxisA += (mine_dist * a); // add a new row to the grid
+		for (int b = 0; b < number_of_objects; b++) { // for each increment along the a plane, add mines along b plane
+			offsetAxisB += (mine_dist * b);           // for each new row populate the column
+
+			/** Now convert the grid (a,b) to the relavenat formation ie. (x,y) or (z,y) etc **/
+			auto ojxyz = xwi_determine_mine_formation_position(oj, objectPosX, objectPosY, objectPosZ, offsetAxisA, offsetAxisB);
+
+			p_object pobj;
+			strcpy_s(pobj.name, xwi_determine_space_object_name(objectNameSet, object_type));
+			pobj.orient = orient;
+			pobj.pos = ojxyz;
+			
+			int team = Species_info[sip->species].default_iff;
+
+			if (number_of_objects > 1) {
+				auto team_name = "Hostile";
+				int index = iff_lookup(team_name);
+				if (index >= 0)
+					team = index;
+				else
+					Warning(LOCATION, "Could not find iff %s", team_name);
+			}
+
+			pobj.arrival_cue = Locked_sexp_true;
+			pobj.arrival_location = ARRIVE_AT_LOCATION;
+			pobj.departure_cue = Locked_sexp_false;
+			pobj.departure_location = DEPART_AT_LOCATION;
+			
+			pobj.ai_class = sip->ai_class;
+			pobj.warpin_params_index = sip->warpin_params_index;
+			pobj.warpout_params_index = sip->warpout_params_index;
+			pobj.ship_max_shield_strength = sip->max_shield_strength;
+			pobj.ship_max_hull_strength = sip->max_hull_strength;
+			Assert(pobj.ship_max_hull_strength > 0.0f); // Goober5000: div-0 check (not shield because we might not have one)
+			pobj.max_shield_recharge = sip->max_shield_recharge;
+			pobj.replacement_textures = sip->replacement_textures; // initialize our set with the ship class set, which may be empty
+			pobj.score = sip->score;
+
+			pobj.team = team;
+			pobj.initial_velocity = 0;
+
+			Parse_objects.push_back(pobj);
```

Comment:
The object will need its subsystems handled as well:
```
switch (oj->objectType) {
	case XWMObjectType::oj_Mine1:
	case XWMObjectType::oj_Mine2:
	case XWMObjectType::oj_Mine3:
	case XWMObjectType::oj_Mine4:
		pobj.subsys_index = Subsys_index;
		int this_subsys = allocate_subsys_status();
		pobj.subsys_count++;
		strcpy_s(Subsys_status[this_subsys].name, NOX("Pilot"));

		for (int n = 0; n < sip->n_subsystems; n++) {
			auto subsys = &sip->subsystems[n];
			if (subsys->type == SUBSYSTEM_TURRET) {
				this_subsys = allocate_subsys_status();
				pobj.subsys_count++;
				strcpy_s(Subsys_status[this_subsys].name, sip->subsystems[n].name);

				for (int bank = 0; bank < MAX_SHIP_PRIMARY_BANKS; bank++) {
					if (subsys->primary_banks[bank] >= 0) {
						Subsys_status[this_subsys].primary_banks[bank] = mine_laser_index;
					}
				}
			}
		}
		break;
	default:
		break;
}
```
And `mine_laser_index` will need to be determined near the top of the function - this will change the ion to the laser.